fastavro/reader.py

```diff
@@ -141,16 +141,15 @@ def read_boolean(fo, writer_schema=None, reader_schema=None):
 
 
 def parse_timestamp(data, resolution):
-    return datetime.datetime.fromtimestamp(data / resolution).replace(
-        microsecond=data % resolution)
+    return datetime.datetime.fromtimestamp(data / resolution)
 
 
 def read_timestamp_millis(data, writer_schema=None, reader_schema=None):
-    return parse_timestamp(data, 1000)
+    return parse_timestamp(data, 1000.0)
```

Comment:
Unlike Python 3, In Python 2 division operator (at parse_timestamp) by integer will output integer and loose fractional seconds.. The float here forces division operator to do a float division in Python 2, and not impacting Python 3.